### PR TITLE
create the IBMCLoudPlatform type for the ingress operator try2

### PR DESCRIPTION
**Pull request by AndreaM12345**
created new PR to indicate the changes as it seems old PR (https://github.com/openshift/api/pull/536) is pulling in too many files due to the api repo having changed quite a bit since that PR was opened.

### CODE REVIEW

**Comment by AndreaM12345** [comment]
cc @csrwng @derekwaynecarr using this PR as it seems cleaner the changes than https://github.com/openshift/api/pull/536

**Comment by AndreaM12345** [comment]
are there some logs to indicate why the verify-client-go test is failing?  it just seems to say 
"Submitted failure event to sentry (id=26e749ea8a894ce4ab5a6459b6936d88)
error: could not run steps: step verify-client-go failed: test "verify-client-go" failed: the pod ci-op-9xyqn8tc/verify-client-go failed after 8m5s (failed containers: ): ContainerFailed one or more containers exited"
But doesn't give detail on what might have cause the container to exit or logs from the container

**Comment by AndreaM12345** [comment]
https://github.com/openshift/origin/pull/24346 is the corresponding openshift/origin PR

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -94,6 +94,20 @@ const (
 
 	// OvirtPlatformType represents oVirt/RHV infrastructure.
 	OvirtPlatformType PlatformType = "oVirt"
+
+	// IBMCloudPlatformType represents IBM CLoud infrastructure.
```
Comment (ironcladlou) [inline config/v1/types_infrastructure.go]: typo: capital L in Cloud

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -94,6 +94,20 @@ const (
...
+)
+
+// IBMCloudProviderType is a specific supported IBM Cloud provider cluster type
+type IBMCloudProviderType string
+
+const (
+	// Classic  means that the IBM Cloud cluster is using classic infrastructure
+	IBMCloudProviderTypeClassic IBMCloudProviderType = "Classic"
+
+	// VPC  means that the IBM Cloud cluster is using VPC infrastructure
```
Comment (ironcladlou) [inline config/v1/types_infrastructure.go]: Extraneous space after VPC

File: config/v1/types_infrastructure.go [hunk 3]
```diff
@@ -230,6 +248,18 @@ type OvirtPlatformStatus struct {
...
 
+//IBMCloudPlatformStatus holds the current status of the IBMCloud infrastructure provider.
+type IBMCloudPlatformStatus struct {
+	// Location is where the cluster has been deployed
+	Location string `json:"location,omitempty"`
+
+	// ResourceGroupName is the Resource Group for new IBMCloud resources created for the cluster.
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+
+	// ProviderType  Indicates the type of cluster that was created
```
Comment (ironcladlou) [inline config/v1/types_infrastructure.go]: Extraneous space and capital I

**Comment by csrwng** [comment]
@AndreaM12345 you need to re-run `make update` to update generated files

**Comment by derekwaynecarr** [comment]
fields all make sense.

/lgtm
/approve

**Pull request merged**